### ISSUE TITLE
Jupyter extension is outdated

### ISSUE THREAD

**Issue by benz0li** [open-vsx/publish-extensions#740]
## Outdated (MS marketplace > Open VSX version)

1. [ms-toolsai.jupyter](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter) (installs: 74.787M, daysInBetween: -150): 2024.1.1 > 2023.9.100

See also https://github.com/open-vsx/publish-extensions/actions/runs/7839191547

**Comment by benz0li** [comment]
@filiptronicek 2024.1.1 – v2024.1.1 to be precise – corresponds to https://github.com/microsoft/vscode-jupyter/tags.

**Comment by benz0li** [comment]
@amvanbaren Has the long-term viability of https://open-vsx.org \[as intended by the Open VSX Working Group\] been secured?

**Comment by filiptronicek** [comment]
Thanks for the report, @benz0li. I will take a look into what's wrong with Jupyter this time.

What already seems to have changed since last time is that the Microsoft Marketplace now offers platform-specific releases for Jupyter, which we should also have in ovsx.

**Comment by filiptronicek** [comment]
I have manually bumped Jupyter to 2024.5.0

https://open-vsx.org/extension/ms-toolsai/jupyter

cc @benz0li

**Comment by benz0li** [comment]
> I have manually bumped Jupyter to 2024.5.0

@filiptronicek Thank you!

The current version of VS Code (1.88) supports up to Jupyter extension v2024.3.1.

I assume Jupyter extension v2024.5.0 will be supported by the next version of VS Code (1.89), which will be released by the end of this week.

**Comment by benz0li** [comment]
IMHO no need to also provide v2024.3.1 on Open VSX.

**Comment by filiptronicek** [comment]
@benz0li you're absolutely right, me publishing May's version was a mistake by accidentally building from `main`. I published `v2024.3.1` now as well so both should be covered.

**Comment by benz0li** [comment]
> @benz0li you're absolutely right, me publishing May's version was a mistake by accidentally building from `main`. I published `v2024.3.1` now as well so both should be covered.

I get the following error now:

```
Installing extension 'ms-toolsai.jupyter'...
error parent:178 Uncaught exception: Invalid comment length. Expected: 63. Found: 0. Are there extra bytes at the end of the file? Or is the end of central dir signature `PK☺☻` in the comment?
error parent:178 Invalid: Invalid comment length. Expected: 63. Found: 0. Are there extra bytes at the end of the file? Or is the end of central dir signature `PK☺☻` in the comment?
```

**Comment by wjandrea** [comment]
> I get the following error now:
> 
> ```
> Installing extension 'ms-toolsai.jupyter'...
> error parent:178 Uncaught exception: Invalid comment length. Expected: 63. Found: 0. Are there extra bytes at the end of the file? Or is the end of central dir signature `PK☺☻` in the comment?
> error parent:178 Invalid: Invalid comment length. Expected: 63. Found: 0. Are there extra bytes at the end of the file? Or is the end of central dir signature `PK☺☻` in the comment?
> ```

I'm getting similar errors (using **Install from VSIX...**):

v2024.3.1
```
Invalid comment length. Expected: 63. Found: 0. Are there extra bytes at the end of the file? Or is the end of central dir signature `PK☺☻` in the comment?
```

v2024.5.0
```
Invalid comment length. Expected: 65. Found: 0. Are there extra bytes at the end of the file? Or is the end of central dir signature `PK☺☻` in the comment?
```

I tried uninstalling and reinstalling the extension, but no luck. I had to revert to v2023.9.100.

If it's relevant, I noticed that the two VSIXs aren't valid ZIP files. `file` just says "data".

**Comment by cosing** [comment]
I noticed some strange bytes appearing in the header and at the end of the file.
![image](https://github.com/open-vsx/publish-extensions/assets/63569926/f738fe3a-c52b-49ec-9827-4d387499103e)
![image](https://github.com/open-vsx/publish-extensions/assets/63569926/7c1e5f4f-b974-4ffc-9c09-f21a6bfacb95)

I use imhex to remove them，this file can be opened as a zip now，and install,it works.

I'm not sure what consequences this will have, but at least it looks like the installation was successful.

**Comment by filiptronicek** [comment]
Thank you all for the reports. This seems to be because of an issue when publishing, since the vsix I built for release seems to install fine. If you'd like to install that manually, I uploaded it below. I also raised https://github.com/EclipseFdn/open-vsx.org/issues/2537, which should help us with re-uploading and fixing this.

[ms-toolsai.jupyter-2024.3.1.vsix.zip](https://github.com/open-vsx/publish-extensions/files/15160290/ms-toolsai.jupyter-2024.3.1.vsix.zip)

**Comment by benz0li** [comment]
> Thank you all for the reports. This seems to be because of an issue when publishing, since the vsix I built for release seems to install fine. If you'd like to install that manually, I uploaded it below. I also raised [EclipseFdn/open-vsx.org#2537](https://github.com/EclipseFdn/open-vsx.org/issues/2537), which should help us with re-uploading and fixing this.

@filiptronicek Why do you zip the VSIX again? These files are in fact already zip files.

I.e. with [the unzipped] [ms-toolsai.jupyter-2024.3.1.vsix.zip](https://github.com/open-vsx/publish-extensions/files/15160290/ms-toolsai.jupyter-2024.3.1.vsix.zip) I cannot change the file suffix from `.vsix` to `.zip` and unzip.  
ℹ️ This works perfectly fine with [ms-toolsai.jupyter-2023.9.100.vsix](https://open-vsx.org/api/ms-toolsai/jupyter/2023.9.100/file/ms-toolsai.jupyter-2023.9.100.vsix).

**Comment by benz0li** [comment]
@filiptronicek https://github.com/open-vsx/publish-extensions/issues/740#issuecomment-2084681127 is still valid.

👉 There is something wrong with the VSIX file.

**Comment by filiptronicek** [comment]
@benz0li of course it does not work, because I accidentally uploaded the file from open-vsx.org I was comparing it with 🤦. I updated the comment with the correct one, which I made sure you can unzip.

**Comment by benz0li** [comment]
> I updated the comment with the correct one, which I made sure you can unzip.

I can confirm that it works now.

**Comment by filiptronicek** [comment]
We have removed the faulty versions and replaced `v2024.3.1` with a correct package. I will publish `v2024.5.0` when it actually becomes stable.

**Comment by benz0li** [comment]
Code - OSS v1.89.0 was released yesterday: https://github.com/microsoft/vscode/releases/tag/1.89.0

@filiptronicek Could you please publish `v2024.4.0` of the Jupyter extension?  
ℹ️ That is the latest version compatible with Code - OSS v1.89.

Thank you for all the hard work.

**Comment by filiptronicek** [comment]
> @filiptronicek Could you please publish v2024.4.0 of the Jupyter extension?

@benz0li I have just published the extension @ v2024.4.0 and fixed the build script to publish new releases automatically and correctly. I will continue to look into how we can provide platform-specific releases via this repo next.

**Comment by benz0li** [comment]
Thank you @filiptronicek.
